php/src/Verifier.php

```diff
@@ -189,8 +190,11 @@ public function requireHeaders(array $headers): IVerifier
      */
     public function verify(string $signature): void
     {
-        $jws = $this->serializerManager
-            ->unserialize($signature);
+        try {
+            $jws = $this->serializerManager->unserialize($signature);
+        } catch (Exception $e) {
+            throw new InvalidSignatureException('Failed to parse JWS: ' . $e->getMessage());
```

Comment:
May be worth chaining the original exception here:

```suggestion
            throw new InvalidSignatureException('Failed to parse JWS: ' . $e->getMessage(), 0, $e);
```